lib/gui/satin_multicolor/colorize.py

```diff
@@ -0,0 +1,265 @@
+        if index == len(main_sizer.GetChildren()) - 1:
+            last_sizer = main_sizer.GetChildren()[-2].GetSizer().GetChildren()
+            last_sizer[2].GetWindow().Show(False)
+            sizer.GetChildren()[2].GetWindow().Show()
+        index = max(0, (index - 1))
+        if index == 0:
+            previous_first = main_sizer.GetChildren()[0].GetSizer().GetChildren()
+            previous_first[0].GetWindow().Show()
+            sizer.GetChildren()[0].GetWindow().Show(False)
+
+        main_sizer.Detach(sizer)
+        main_sizer.Insert(index, sizer, 0, wx.EXPAND | wx.ALL, 10)
+        self.FitInside()
+        self._update()
+        self.Layout()
+
+    def _remove_color(self, event):
+        sizer = event.GetEventObject().GetContainingSizer()
+        sizer.Clear(True)
+        self.color_sizer.Remove(sizer)
+        self.FitInside()
+
+        self._update_colors()
+        self._update()
+
+    def _on_update_equidistance(self, event=None):
+        if self.equististance.GetValue():
+            self.monochrome_width_label.Enable(True)
+            self.monochrome_width.Enable(True)
+            self._set_widget_status(False)
+            self._update_colors()
+        else:
+            self.monochrome_width_label.Enable(False)
+            self.monochrome_width.Enable(False)
+            self._set_widget_status(True)
+        self._update()
+
+    def _set_widget_status(self, status):
+        for color in self.color_sizer.GetChildren():
+            inner_sizer = color.GetSizer()
+            for color_widget in inner_sizer:
+                widget = color_widget.GetWindow()
+                if isinstance(widget, wx.SpinCtrlDouble):
+                    widget.Enable(status)
+
+    def _set_widget_width_value(self, value, margin=0):
+        first = True
+        for color in self.color_sizer.GetChildren():
+            inner_sizer = color.GetSizer()
+            for color_widget in inner_sizer:
+                widget = color_widget.GetWindow()
+                if first and widget.Label == "↑":
+                    inner_sizer.Hide(widget)
+                    first = False
+                if isinstance(widget, wx.SpinCtrlDouble):
+                    widget.SetValue(value)
+                    widget.GetNextSibling().SetValue(margin)
+                    break
+
+    def get_total_width(self):
+        width = 0
+        colors = self.color_sizer.GetChildren()
+        for color in colors:
+            inner_sizer = color.GetSizer()
+            for color_widget in inner_sizer:
+                widget = color_widget.GetWindow()
+                if isinstance(widget, wx.SpinCtrlDouble):
+                    width += widget.GetValue()
+        last_margin = inner_sizer.GetChildren()[3].GetWindow().GetValue()
+        width -= last_margin
+        return round(width, 2)
+
+    def _update(self, event=None):
+        width = self.get_total_width()
+        self.total_width.SetLabel(_(f"Total width: { width } %"))
```

Comment:
Does this work? I'd expect it to resolve to e.g. `_("Total width 104 %")` then fail to translate because there won't be a translation key for that number. Unless there's some magic under the hood with _ that makes it work, it may have to become `_("total width: {width} %").format(width=width)`. 
(c.f. https://stackoverflow.com/questions/49797658/how-to-use-gettext-with-python-3-6-f-strings)

Also, should it have the space between the curly brace and the percent sign? I usually don't see spaces between numbers and the percent sign in English.

Reply:
Oh, thanks. What was I thinking :)